### PR TITLE
[CBRD-23709] error message is displayed incorrectly in 'MERGE' clause

### PR DESCRIPTION
**Pull request by shparkcubrid**
http://jira.cubrid.org/browse/CBRD-23709

fix to display 'Multiple rows in source table match the same row in destination table.' error when there are no existing errors

### CODE REVIEW

File: src/query/query_executor.c [hunk 1]
```diff
@@ -4719,7 +4719,8 @@ qexec_groupby (THREAD_ENTRY * thread_p, XASL_NODE * xasl, XASL_STATE * xasl_stat
     result = (gbstate.state == NO_ERROR || gbstate.state == SORT_PUT_STOP) ? NO_ERROR : ER_FAILED;
 
     /* check merge result */
-    if (XASL_IS_FLAGED (xasl, XASL_IS_MERGE_QUERY) && list_id->tuple_cnt != gbstate.input_recs)
+    if (gbstate.state == NO_ERROR && XASL_IS_FLAGED (xasl, XASL_IS_MERGE_QUERY)
```
Comment (hgryoo) [inline src/query/query_executor.c:4722]: `gbstate.state == SORT_PUT_STOP` seems not to be considered as an error too.

```suggestion
    if (result == NO_ERROR && XASL_IS_FLAGED (xasl, XASL_IS_MERGE_QUERY)
```

**Comment by cubridci** [comment]
Build comment file: 
--------------
#### File: src/query/query_executor.c (1 incorrect indentation(s) found)
```
  4722     if (result == NO_ERROR && XASL_IS_FLAGED (xasl, XASL_IS_MERGE_QUERY) && list_id->tuple_cnt != gbstate.input_recs)
```
--------------
**Please run 'indent -l120 -lc120' to correct the indentation**

--------------

File: src/query/query_executor.c (hunk 1 above)
Comment (shparkcubrid) [reply src/query/query_executor.c]: thanks

File: src/query/query_executor.c [hunk 2]
```diff
@@ -4719,7 +4719,8 @@ qexec_groupby (THREAD_ENTRY * thread_p, XASL_NODE * xasl, XASL_STATE * xasl_stat
     result = (gbstate.state == NO_ERROR || gbstate.state == SORT_PUT_STOP) ? NO_ERROR : ER_FAILED;
 
     /* check merge result */
-    if (XASL_IS_FLAGED (xasl, XASL_IS_MERGE_QUERY) && list_id->tuple_cnt != gbstate.input_recs)
+    if (result == NO_ERROR && XASL_IS_FLAGED (xasl, XASL_IS_MERGE_QUERY)
```
Comment (shparkcubrid) [inline src/query/query_executor.c:4722]:
```suggestion
    if (result == NO_ERROR && XASL_IS_FLAGED (xasl, XASL_IS_MERGE_QUERY) && list_id->tuple_cnt != gbstate.input_recs)
```

File: src/query/query_executor.c [hunk 3]
```diff
@@ -4719,7 +4719,8 @@ qexec_groupby (THREAD_ENTRY * thread_p, XASL_NODE * xasl, XASL_STATE * xasl_stat
...
+	&& list_id->tuple_cnt != gbstate.input_recs)
```
Comment (shparkcubrid) [inline src/query/query_executor.c:4723]:
```suggestion
```

**Pull request merged**